packages/common/src/context/commentsContext.tsx

```diff
@@ -23,20 +20,33 @@ import {
   useGetCurrentUserId,
   useGetTrackById
 } from '../api'
+import { usePaginatedQuery } from '../audius-query'
+import { ID, Status } from '../models'
 import { tracksActions } from '../store/pages/track/lineup/actions'
 import { playerSelectors } from '../store/player'
+import { Nullable } from '../utils'
 
 export enum CommentSortMethod {
   top = 'top',
   newest = 'newest',
   timestamp = 'timestamp'
 }
 
+type CommentSectionProviderProps = {
+  entityId: ID
+  entityType?: EntityType.TRACK
+
+  // These are optional because they are only used on mobile
+  // and provided for the components in CommentDrawer
+  replyingToComment?: Comment
+  setReplyingToComment?: (comment: Comment) => void
+  editingComment?: Comment
+  setEditingComment?: (comment: Comment) => void
+}
```

Comment:
I think optionals is the most practical way to go but the TS purist part of me wonders if a union type + type narrowing would be good too? Probably not a practical type to use tho haha

Reply:
Oh I kinda like that, could have a separate useCurrentCommentSection on mobile that does the type narrowing 🤔

let me think on this a bit